pkgs/by-name/he/hello-cpp/package.nix

```diff
@@ -0,0 +1,11 @@
+{
+  cmake,
+  ninja,
+  stdenv,
+}:
+
+stdenv.mkDerivation {
+  name = "hello-cpp";
+  src = ./src;
+  nativeBuildInputs = [ cmake ninja ];
```

Comment:
We may still want a meta attribute, see https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/by-name/he/hello/package.nix#L45 (at least `description`, `mainProgram`, `maintainers` and `platforms`)

Reply:
Any suggestions for who I can put down as a maintainer? My sense is that it should belong to whoever maintains the stdenv in a sense, since if it's broken it would fall to the stdenv/compilers to figure out why. I can't realistically volunteer myself unfortunately.

Reply:
Is `mainProgram` necessary if the package name is the same as the binary name? I thought not.

Reply:
For `platforms` I would hope all of them but I would guess not all platforms support this (for example bare metal etc? So I don't know what set to put there).

Reply:
> Any suggestions for who I can put down as a maintainer? My sense is that it should belong to whoever maintains the stdenv in a sense, since if it's broken it would fall to the stdenv/compilers to figure out why. I can't realistically volunteer myself unfortunately.

But you won't be notified if the `stdenv` or something is broken. Writing maintainers here just means that you are responsible for this package. As the [contribution doc](https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes) suggests, new package must have a maintainer attribute.

> Is `mainProgram` necessary if the package name is the same as the binary name? I thought not.

Yes.

Reply:
Proposal, thoughts?
```nix
  meta = {
    description = "A basic sanity check that C++ and cmake infrastructure is working";
    platforms = lib.platforms.all;
    maintainers = stdenv.cc.meta.maintainers or [];
  };
```

This choice puts it on the cc wrapper maintainers and would reflect the choice of stdenv: `pkgsLLVM.hello-cpp` etc.

Reply:
Looks okay, except `A` should be removed from description

Reply:
Thanks for the quick iteration on this - I've got to go AFK now but will respond in a few hours or tomorrow if there are further comments.